src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.openwifirrm.ucentral.models;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association;
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association.Rate;
+
+/**
+ * Aggregation model for State aggregation. Only contains info useful for analysis.
+ */
+public class AggregatedState {
+
+	public static class AggregatedRate {
+		public long bitRate;
+		public int chWidth;
+		public List<Integer> mcs;
+
+		/** Constructor with no args */
+		public AggregatedRate() {
+			mcs = new ArrayList<>();
+		}
+
+		/** Add a Rate to the AggregatedRate */
+		public void add(Rate rate) {
+			if (rate == null) {
+				return;
+			}
+			if (mcs == null || mcs.isEmpty()) {
```

Comment:
Can `mcs` ever become null? If so, you need to initialize it before line 43 `mcs.add(rate.mcs)`